integration_tests/grpc_rest/dclupgrade/helpers.go

```diff
@@ -0,0 +1,332 @@
+	require.NoError(suite.T, err)
+
+	// Check upgrade is proposed
+	proposedUpgrade, err := GetProposedUpgrade(suite, proposeUpgradeMsg.Plan.Name)
+	require.NoError(suite.T, err)
+	require.Equal(suite.T, proposeUpgradeMsg.Creator, proposedUpgrade.Creator)
+	require.Equal(suite.T, proposeUpgradeMsg.Plan, proposedUpgrade.Plan)
+
+	// Get all proposed upgrades
+	_, err = GetProposedUpgrades(suite)
+	require.NoError(suite.T, err)
+
+	// Get all approved upgrades
+	approvedUpgrades, err := GetApprovedUpgrades(suite)
+	require.NoError(suite.T, err)
+	require.NotContains(suite.T, approvedUpgrades, proposedUpgrade)
+
+	// another trustee approves upgrade
+	approveUpgradeMsg := NewMsgApproveUpgrade(bobAccount.Address, proposeUpgradeMsg.Plan.Name)
+	_, err = suite.BuildAndBroadcastTx([]sdk.Msg{approveUpgradeMsg}, bobName, bobAccount)
+	require.NoError(suite.T, err)
+
+	// Check upgrade is approved
+	approvedUpgrade, err := GetApprovedUpgrade(suite, proposeUpgradeMsg.Plan.Name)
+	require.NoError(suite.T, err)
+	require.Equal(suite.T, proposeUpgradeMsg.Creator, approvedUpgrade.Creator)
+	require.Equal(suite.T, proposeUpgradeMsg.Plan, approvedUpgrade.Plan)
+
+	// Get all proposed upgrades
+	proposedUpgrades, err := GetProposedUpgrades(suite)
+	require.NoError(suite.T, err)
+	require.NotContains(suite.T, proposedUpgrades, proposedUpgrade)
+
+	// Get all approved upgrades
+	approvedUpgrades, err = GetApprovedUpgrades(suite)
+	require.NoError(suite.T, err)
+	require.Contains(suite.T, approvedUpgrades, *approvedUpgrade)
+}
+
+/* Error cases */
+
+func ProposeUpgradeByNonTrustee(suite *utils.TestSuite) {
+	// Alice and Bob are predefined Trustees
+	aliceName := testconstants.AliceAccount
+	aliceKeyInfo, err := suite.Kr.Key(aliceName)
+	require.NoError(suite.T, err)
+	aliceAccount, err := test_dclauth.GetAccount(suite, aliceKeyInfo.GetAddress())
+	require.NoError(suite.T, err)
+
+	bobName := testconstants.BobAccount
+	bobKeyInfo, err := suite.Kr.Key(bobName)
+	require.NoError(suite.T, err)
+	bobAccount, err := test_dclauth.GetAccount(suite, bobKeyInfo.GetAddress())
+	require.NoError(suite.T, err)
+
+	// register new account without Trustee role
+	nonTrusteeAccountName := utils.RandString()
+	nonTrusteeAccount := test_dclauth.CreateAccount(
+		suite,
+		nonTrusteeAccountName,
+		dclauthtypes.AccountRoles{dclauthtypes.CertificationCenter},
+		1,
+		aliceName,
+		aliceAccount,
+		bobName,
+		bobAccount,
+		testconstants.Info,
+	)
+	require.NotContains(suite.T, nonTrusteeAccount.Roles, dclupgradetypes.UpgradeApprovalRole)
+
+	// try to add proposeUpgradeMsg
+	proposeUpgradeMsg := NewMsgProposeUpgrade(nonTrusteeAccount.Address, utils.RandString(), 100000, utils.RandString())
+	_, err = suite.BuildAndBroadcastTx([]sdk.Msg{proposeUpgradeMsg}, nonTrusteeAccountName, nonTrusteeAccount)
+	require.Error(suite.T, err)
+	require.ErrorIs(suite.T, sdkerrors.ErrUnauthorized, err)
+}
+
+func ApproveUpgradeByNonTrustee(suite *utils.TestSuite) {
```

Comment:
See the comments to `ProposeUpgradeByNonTrustee`.

Reply:
fixed